athema/src/main/scala/higherkindness/athema/expr.scala

```diff
@@ -55,15 +55,16 @@ object Div {
 private[athema] sealed trait ExprInstances {
   implicit def traverseExpr[V]: Traverse[Expr[V, *]] =
     new DefaultTraverse[Expr[V, *]] {
-      def traverse[G[_]: Applicative, A, B](fa: Expr[V, A])(
-          f: A => G[B]): G[Expr[V, B]] = fa match {
-        case v: Var[_, B @unchecked]   => (v: Expr[V, B]).pure[G]
-        case c: Const[_, B @unchecked] => (c: Expr[V, B]).pure[G]
-        case e: Neg[V, A]              => f(e.x) map (Neg(_))
-        case e: Add[V, A]              => (f(e.x), f(e.y)) mapN (Add(_, _))
-        case e: Sub[V, A]              => (f(e.x), f(e.y)) mapN (Sub(_, _))
-        case e: Prod[V, A]             => (f(e.x), f(e.y)) mapN (Prod(_, _))
-        case e: Div[V, A]              => (f(e.x), f(e.y)) mapN (Div(_, _))
+      def traverse[G[_]: Applicative, A, B](
+          fa: Expr[V, A]
+      )(f: A => G[B]): G[Expr[V, B]] = fa match {
+        case v: Var[V, A]   => (v.asInstanceOf[Expr[V, B]]).pure[G]
+        case c: Const[V, A] => (c.asInstanceOf[Expr[V, B]]).pure[G]
+        case e: Neg[V, A]   => f(e.x) map (Neg(_))
+        case e: Add[V, A]   => (f(e.x), f(e.y)) mapN (Add(_, _))
+        case e: Sub[V, A]   => (f(e.x), f(e.y)) mapN (Sub(_, _))
+        case e: Prod[V, A]  => (f(e.x), f(e.y)) mapN (Prod(_, _))
+        case e: Div[V, A]   => (f(e.x), f(e.y)) mapN (Div(_, _))
```

Comment:
As a BTW, are we still too keen on infix syntax? 
```suggestion
        case e: Neg[V, A]   => f(e.x) map (Neg(_))
        case e: Add[V, A]   => (f(e.x), f(e.y)).mapN(Add(_, _))
        case e: Sub[V, A]   => (f(e.x), f(e.y)).mapN(Sub(_, _))
        case e: Prod[V, A]  => (f(e.x), f(e.y)).mapN(Prod(_, _))
        case e: Div[V, A]   => (f(e.x), f(e.y)).mapN(Div(_, _))
```
specially if it saves no parentheses.

Reply:
I'm personally not keen on it, I just left things as they were mainly and how scalafmt left them